odl/test/largescale/space/tensor_space_slow_test.py

```diff
@@ -39,119 +39,119 @@
 
 
 @pytest.fixture(scope="module", ids=spc_ids, params=spc_params)
-def fn(fn_impl, request):
+def tspace(tspace_impl, request):
     spc = request.param
 
     if spc == 'rn':
-        return odl.rn(10 ** 5, impl=fn_impl)
+        return odl.rn(10 ** 5, impl=tspace_impl)
     elif spc == '1d':
-        return odl.uniform_discr(0, 1, 10 ** 5, impl=fn_impl)
+        return odl.uniform_discr(0, 1, 10 ** 5, impl=tspace_impl)
     elif spc == '3d':
         return odl.uniform_discr([0, 0, 0], [1, 1, 1],
-                                 [100, 100, 100], impl=fn_impl)
+                                 [100, 100, 100], impl=tspace_impl)
 
 
-def test_element(fn):
-    x = fn.element()
-    assert x in fn
+def test_element(tspace):
+    x = tspace.element()
+    assert x in tspace
 
-    y = fn.element(inp=[0] * fn.size)
-    assert y in fn
+    y = tspace.element(inp=[0] * tspace.size)
+    assert y in tspace
 
     # Rewrap
-    y2 = fn.element(y)
+    y2 = tspace.element(y)
     assert y2 is y
 
-    w = fn.element(inp=np.zeros(fn.size, fn.dtype))
-    assert w in fn
+    w = tspace.element(inp=np.zeros(tspace.size, tspace.dtype))
+    assert w in tspace
 
 
-def test_zero(fn):
-    assert np.allclose(fn.zero(), 0)
+def test_zero(tspace):
+    assert np.allclose(tspace.zero(), 0)
 
 
-def test_one(fn):
-    assert np.allclose(fn.one(), 1)
+def test_one(tspace):
+    assert np.allclose(tspace.one(), 1)
 
 
-def test_ndarray_init(fn):
-    x0 = np.arange(fn.size)
-    x = fn.element(x0)
+def test_ndarray_init(tspace):
+    x0 = np.arange(tspace.size)
+    x = tspace.element(x0)
 
     assert all_almost_equal(x0, x)
 
 
-def test_getitem(fn):
-    indices = np.random.randint(0, fn.size - 1, 5)
+def test_getitem(tspace):
+    indices = np.random.randint(0, tspace.size - 1, 5)
 
-    x0 = np.arange(fn.size)
-    x = fn.element(x0)
+    x0 = np.arange(tspace.size)
+    x = tspace.element(x0)
 
     for index in indices:
         assert x[index] == index
 
 
-def test_setitem(fn):
-    indices = np.random.randint(0, fn.size - 1, 5)
+def test_setitem(tspace):
+    indices = np.random.randint(0, tspace.size - 1, 5)
 
-    x0 = np.arange(fn.size)
-    x = fn.element(x0)
+    x0 = np.arange(tspace.size)
+    x = tspace.element(x0)
```

Comment:
What was the decision on converting flat arrays to rectangular arrays? Since you do it here.